### DISCUSSION TITLE
Welcome to stable-diffusion.cpp Discussions!

### DISCUSSION THREAD

**Discussion by leejet** [leejet/stable-diffusion.cpp#29] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by GDani31** [comment]
Thanks for doing this! I like it how easy to compile and use it is and the tool for convertion. Sadly for me it is still slow but the progress is very good. some days ago i had like 50 sec per step and now its 35 sec per step!

**Comment by Jonathhhan** [comment]
Thanks, very nice project (looking forward for GPU support).
Its the first (and only) c++ stable diffusion implementation, where I was able to integrate it into Open Frameworks:
https://github.com/Jonathhhan/ofStableDiffusionExample/tree/main

**Comment by JohnClaw** [comment]
SD.cpp is my love. I'm making a GUI for it using twinBasic. Btw, are there other AI apps like SD.cpp? For example, Meta_AI_MMS_TTS.cpp.exe or something like Songs_and_Music_AI_Generator.cpp.exe or Meta_AI_SeamlessM4T.cpp.exe?

**Comment by Amin456789** [comment]
Thank you for ur hard work leejet and whoever is involved in improving this!

@johnclaw great! will u release ur gui when it is ready for public?
also johnclaw, there is bark.cpp that they have plan to release Audiocraft for cpp, u should check it out

**Comment by cmp-nct** [comment]
What a huge effort, very valuable to have that codebase.

**Comment by evan-anthony** [comment]
Does anyone have a GUI yet that they have put on the front of this? I haven't seen anything from Automatic1111 offering support for SD.cpp and would like to run this with a GUI

**Comment by JohnClaw** [comment]
> Does anyone have a GUI yet that they have put on the front of this? I haven't seen anything from Automatic1111 offering support for SD.cpp and would like to run this with a GUI

Professional GUI: https://github.com/Cyberhan123/stable-diffusion-desktop

My amateur GUI (supports only LCM Dreamshaper and hasn't been released yet):
![image](https://github.com/leejet/stable-diffusion.cpp/assets/131554379/5345bdb9-efaf-4745-af3b-e21d431e5e40)

**Reply by Amin456789** [reply]
when u released ur gui i would like to give u suggetions to improve it johnclaw, please release it

**Reply by fszontagh** [reply]
I made one too: https://github.com/fszontagh/sd.cpp.gui.wx

Not perfect, and not very tested.
![sd cpp gui wx_queue](https://github.com/leejet/stable-diffusion.cpp/assets/51741446/8de5e3ac-de5c-45cb-8d41-e3469fd434c9)

**Reply by Amin456789** [reply]
its lovely, thank u so much

**Comment by Jonathhhan** [comment]
@eanthony76 my GUI attempt as an Open Frameworks addon: https://github.com/Jonathhhan/ofxStableDiffusion/tree/main

**Comment by Amin456789** [comment]
@leejet could u please give a status progress of merging controlnet, svd and inpainting

**Reply by Green-Sky** [reply]
#131 is merged now

**Comment by phudtran** [comment]
I saw that there is now support for prompts longer than 77 tokens. What is the new upper limit?

**Reply by leejet** [reply]
No upper limit.

**Comment by treksis** [comment]
such a great repo. please support inpainting.

**Comment by waheedi** [comment]
Hello, Waheed is here, I used your repo for a test run on RX5700 i really like the cpp implementation, good job :)

I have seen a performance decrease in comparison with basic pytorch StableDiffusion-webui I get an image in 7.3 seconds using the standard default params, while i get an image in 11.27s using your repo. (but also I see a higher quality images using your imp)

I see that you mentioned faster ggml implementation for ggml_conv_2d that might already help.

Also one more question regarding multi-gpu usage, it currently can detects several GPUs but it does only use one of them, is there a way to use all the CUDA buffers I think it should be possible.

**Comment by DenisMontes** [comment]
Dear community,
I am pleased to announce that I am currently working on a Node.js binding for this project, with plans to extend its integration—alongside GGML—to other programming languages in the near future.
This initiative is part of a broader effort to democratize access to inference and training frameworks, ensuring they are accessible to developers beyond the Python ecosystem.
I look forward to sharing updates on the progress of this binding soon.
